packages/react/src/hooks/useSizeObserver.js

```diff
@@ -0,0 +1,23 @@
+import useResizeObserver from 'use-resize-observer';
+import warning from 'warning';
+
+import { browserSupports } from '../utils/componentUtilityFunctions';
+
+const useSizeObserver = (ref) => {
```

Comment:
What is the purpose of this hook?  is it to measure the size of an element after it has rendered?

And more importantly, since this is replacing the sizeme with this (I think), what are the benefits of this vs sizeme?  ie what was it about sizeme that made it unsuitable and needed to be replaced?